devlab/tests/test_verify_dst_functionality.py

```diff
@@ -0,0 +1,318 @@
+        if servers:
+            for server in servers:
+                self.dst_cloud.novaclient.servers.delete(server.id)
+
+        # delete any cinder volumes
+        cinder_volumes = self.dst_cloud.cinderclient.volumes.list()
+        if cinder_volumes:
+            for volume in cinder_volumes:
+                try:
+                    self.dst_cloud.cinderclient.volumes.delete(volume.id)
+                except CinderClientBaseException:
+                    pprint.pprint("Volume {0:s} failed to delete"
+                                  .format(volume.display_name))
+
+        # delete keypairs after test_keypair test
+        if self.dst_cloud.novaclient.keypairs.list():
+            for key_p in self.dst_cloud.novaclient.keypairs.list():
+                key_p.delete()
+
+        self.release_fips_tenant()
+        self.update_floatip_neutron_quota(
+            self.fip_quota_neutron, TEST_TENANT_NAME)
+
+    def check_vm_ssh_access(self, vm, ip_addr, username, cmd):
+        with settings(host_string=ip_addr, user=username,
+                      key=config.private_key['id_rsa'],
+                      abort_on_prompts=True, connection_attempts=3,
+                      disable_known_hosts=True):
+            try:
+                run(cmd, shell=False)
+                msg = 'VM {name} with ip {addr} is accessible via ssh'
+                return msg.format(name=vm.name, addr=ip_addr), True
+            except SystemExit:
+                msg = 'VM {name} with ip {addr} is not accessible via ssh'
+                return msg.format(name=vm.name, addr=ip_addr), False
+
+    def check_cinder_volume_on_vm(self, vm, ip_addr, username, cmd):
+        with settings(host_string=ip_addr,
+                      user=username,
+                      key=config.private_key['id_rsa'],
+                      shell=config.ssh_vm_shell,
+                      abort_on_prompts=True,
+                      connection_attempts=3,
+                      disable_known_hosts=True):
+            try:
+                for command in cmd:
+                    sudo(command, shell=True)
+                msg = 'Cinder volume is OK and accessible on' \
+                      'VM with name {name} and ip: {addr}'
+                return msg.format(name=vm.name, addr=ip_addr), True
+            except SystemExit:
+                msg = 'Cinder volume access FAIL on' \
+                      'VM with name {name} and ip: {addr}'
+                return msg.format(name=vm.name, addr=ip_addr), False
+
+    def release_fips_tenant(self):
+        fip_tenant_ids = [f['id'] for f in
+                          self.dst_cloud.neutronclient.list_floatingips(
+        )['floatingips']
+            if f['tenant_id'] == self.dst_tenant_id]
+
+        for f_id in fip_tenant_ids:
+            self.dst_cloud.neutronclient.delete_floatingip(f_id)
+        time.sleep(3)
+
+    def update_floatip_neutron_quota(self, quota_value, tenant_name):
```

Comment:
In this suite tenant_name always is TEST_TENANT_NAME. Method update_floatip_neutron_quota can be refactored to always use TEST_TENANT_NAME (not take tenant_name as a argument).